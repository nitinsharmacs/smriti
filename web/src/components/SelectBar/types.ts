import type { SvgIconProps } from '@mui/material';

export type SelectBarControl = {
  icon?: React.ElementType<SvgIconProps>;
  handler: () => void;
};

export interface SelectBarProps {
  open: boolean;
  onClose: () => void;
  controls: SelectBarControl[];
  selections?: number;
}
