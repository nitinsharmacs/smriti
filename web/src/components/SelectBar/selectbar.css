.select-bar {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: var(--secondary-bg-color);
  z-index: 1000;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  color: var(--main-text-fg-color);
  padding: 0em 1.5em;
}

.select-bar button {
  color: var(--main-text-fg-color);
}

.select-bar-display-text {
  font-size: 1.2rem;
  margin-left: 1em;
}

.select-bar-display-text span:last-child {
  margin-left: 0.5em;
}

.select-bar-controls {
  flex-grow: 1;
  display: flex;
  justify-content: flex-end;
}

.select-bar-control {
  margin: 0.5em 0;
}
