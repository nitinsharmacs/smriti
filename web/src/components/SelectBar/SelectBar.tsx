import IconButton from '@mui/material/IconButton';
import ClearIcon from '@mui/icons-material/Clear';
import DeleteIcon from '@mui/icons-material/Delete';
import ShareIcon from '@mui/icons-material/Share';

import './selectbar.css';

const SelectBar = () => {
  return (
    <div className='select-bar'>
      <div>
        <IconButton className='select-bar-close-btn'>
          <ClearIcon />
        </IconButton>
      </div>
      <div className='select-bar-display-text'>
        <span>2</span>
        <span>Selected</span>
      </div>
      <div className='select-bar-controls'>
        <div className='select-bar-control'>
          <IconButton className='select-bar-close-btn'>
            <DeleteIcon />
          </IconButton>
          <IconButton className='select-bar-close-btn'>
            <ShareIcon />
          </IconButton>
        </div>
      </div>
    </div>
  );
};

export default SelectBar;
