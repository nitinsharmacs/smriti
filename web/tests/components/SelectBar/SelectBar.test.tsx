import { render } from '@testing-library/react';
import SelectBar from 'src/components/SelectBar/SelectBar';
import { describe, expect, it } from 'vitest';

describe('SelectBar', () => {
  it('should match snapshot', () => {
    const { container } = render(<SelectBar />);
    expect(container).toMatchSnapshot();
  });
});
