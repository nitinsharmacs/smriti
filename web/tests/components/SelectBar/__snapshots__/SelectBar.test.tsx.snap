// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectBar > should match snapshot 1`] = `
<div>
  <div
    class="select-bar"
  >
    <div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium select-bar-close-btn css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
    <div
      class="select-bar-display-text"
    >
      <span>
        2
      </span>
      <span>
        Selected
      </span>
    </div>
    <div
      class="select-bar-controls"
    >
      <div
        class="select-bar-control"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium select-bar-close-btn css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium select-bar-close-btn css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="ShareIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92s2.92-1.31 2.92-2.92-1.31-2.92-2.92-2.92"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
